lib/src/entrypoint.dart

```diff
@@ -81,71 +79,68 @@ class Entrypoint {
   final SystemCache cache;
 
   /// Whether this entrypoint exists within the package cache.
-  bool get isCached => root.dir != null && p.isWithin(cache.rootDir, root.dir);
+  bool get isCached => root.dir != null && p.isWithin(cache.rootDir, root.dir!);
 
   /// Whether this is an entrypoint for a globally-activated package.
   final bool isGlobal;
 
   /// The lockfile for the entrypoint.
   ///
   /// If not provided to the entrypoint, it will be loaded lazily from disk.
-  LockFile get lockFile {
-    if (_lockFile != null) return _lockFile;
+  LockFile get lockFile => _lockFile ??= _loadLockFile();
 
+  LockFile _loadLockFile() {
     if (!fileExists(lockFilePath)) {
-      _lockFile = LockFile.empty();
+      return _lockFile = LockFile.empty();
     } else {
-      _lockFile = LockFile.load(lockFilePath, cache.sources);
+      return _lockFile = LockFile.load(lockFilePath, cache.sources);
     }
-
-    return _lockFile;
   }
 
-  LockFile _lockFile;
+  LockFile? _lockFile;
 
   /// The package graph for the application and all of its transitive
   /// dependencies.
   ///
   /// Throws a [DataError] if the `.dart_tool/package_config.json` file isn't
   /// up-to-date relative to the pubspec and the lockfile.
-  PackageGraph get packageGraph {
-    if (_packageGraph != null) return _packageGraph;
+  PackageGraph get packageGraph => _packageGraph ??= _createPackageGraph();
 
+  PackageGraph _createPackageGraph() {
     assertUpToDate();
     var packages = {
       for (var id in lockFile.packages.values) id.name: cache.load(id)
     };
     packages[root.name] = root;
 
-    _packageGraph = PackageGraph(this, lockFile, packages);
-    return _packageGraph;
+    return PackageGraph(this, lockFile, packages);
   }
 
-  PackageGraph _packageGraph;
+  PackageGraph? _packageGraph;
 
   /// Where the lock file and package configurations are to be found.
   ///
   /// Global packages (except those from path source)
   /// store these in the global cache.
-  String get _configRoot =>
+  String? get _configRoot =>
       isCached ? p.join(cache.rootDir, 'global_packages', root.name) : root.dir;
```

Comment:
```suggestion
String get _configRoot {
    if (isCached) {
      return p.join(cache.rootDir, 'global_packages', root.name);
    }
    final dir = root.dir;
    if (dir != null) {
      return dir;
    }
    throw UnsupportedError(
      'Unsupported operation on Entrypoint with in-memory package.',
    );
  }
```

Reply:
As discussed, I prefer to do it at a higher level.